src/domain/proof-of-liabilities/generate-liabilities-tree.ts

```diff
@@ -0,0 +1,33 @@
+import { getHash } from "@domain/proof-of-liabilities/generate-leaves"
+import { toSats } from "@domain/bitcoin"
+
+const getSum = (
+  val1: CurrencyBaseAmount,
+  val2: CurrencyBaseAmount,
+): CurrencyBaseAmount => {
+  return toSats(parseInt(val1.toString()) + parseInt(val2.toString()))
+}
+export const GenerateLiabilitiesTree = (leaves: TreeNode[]) => {
```

Comment:
@bodymindarts This method assumes that the number of leaves are a power of 2, so that it forms a perfect binary tree. What is the correct way of enforcing this? Should we check and return an error if it is not?